PROTOCOL.md

```diff
@@ -1046,6 +1048,31 @@ When Row Tracking is enabled (when the table property `delta.enableRowTracking`
   In particular, writers should set `delta.rowTracking.preserved` in the `tags` of the `commitInfo` action to `true` if no rows are updated or copied.
   Writers should set that flag to false otherwise.
 
+# Clustered Table
+
+The Clustered Table feature facilitates the physical clustering of rows that share similar values on a predefined set of clustering columns.
+This enhances query performance when selective filters are applied to these clustering columns through data skipping.
+Clustering columns must be specified during the initial definition of a clustered table, and they can be modified after the table has been created.
+
+A table is defined as a clustered table through the following criteria:
+- When the feature `clustering` exists in the table `protocol`'s `writerFeatures`, then we say that the table is a clustered table.
+  The feature `domainMetadata` is required in the table `protocol`'s `writerFeatures`.
+
+Enablement:
+- The table must be on Writer Version 7.
+- The feature `clustering` must exist in the table `protocol`'s `writerFeatures`.
+
+## Writer Requirements for Clustered Table
+
+When the Clustered Table is supported (when the `writerFeatures` field of a table's `protocol` action contains `clustering`), then:
+- Writers must write out [per-file statistics](#per-file-statistics) and per-column statistics for clustering columns in `add` action. 
+  Failure to collect per-column statistics for clustering columns will result in an error when defining a clustered table or making changes to the clustering columns.
+- Writers must track clustering column names in a `domainMetadata` action with `delta.clustering` as the `domain` and a `configuration` containing all clustering column names.
+  If [Column Mapping](#column-mapping) is enabled, the physical column names should be used.
+- When a clustering implementation clusters files, writers must incorporate a `tag`  with `CLUSTERED_BY` as the key and the name of the clustering implementation as the corresponding value in `add` action.
```

Comment:
Newly inserted files have not been clustered yet, so any implementation can "claim" them if it gets there first? I guess that makes sense, but wanted to double check that this is the intent. Should we update the spec to talk about this? That the writer is not required to cluster any given file at any particular time (but still must write accurate stats); but if/when it chooses to cluster a given file, then it must store the CLUSTERED_BY tag?

Reply:
+1

Reply:
```suggestion
- When a clustering implementation clusters files, writers must incorporate a `tag`  with `CLUSTERED_BY` as the key and the name of the clustering implementation as the corresponding value in the `add` actions for the clustered files.
```

Reply:
fixed